test/integration/master/kube_apiserver_test.go

```diff
@@ -85,6 +85,43 @@ func TestRun(t *testing.T) {
 	}
 }
 
+func TestReadyz(t *testing.T) {
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--post-start-hook-grace-period", "10s"}, framework.SharedEtcd())
```

Comment:
For better testing you want a `DontWaitForHealthz` in `TestServerInstanceOptions`.

Reply:
When the grace period is enabled, healthz returns true immediately, so we actually don't need to explicitly configure that.